File: mobile/lib/src/core/lesson_icons.dart
/// Slug → [IconData] for the lesson-node glyph set.
///
/// The set itself is defined once in `shared/lessonIcons.js` (the admin panel
/// renders it, the backend validates against it, the web app maps it to
/// lucide); this is its Flutter half. Adding an icon means touching all three
/// — a slug missing here falls back to the caller's default glyph, which is
/// safe but wrong, so keep the lists in step.
///
/// The glyphs themselves now come from lucide, the same source the web
/// bundles (core/lucide_svg.dart) — Material Rounded was the closest
/// available match and it still was not close: filled and chunky against
/// lucide's thin geometry, so the same lesson looked like two different
/// products side by side. The Material map below survives as the fallback
/// for a slug the generated set does not carry.
library;

import 'package:flutter/material.dart';

import '../widgets/lucide_icon.dart';

const Map<String, IconData> _lessonIcons = {
  // Акча жана төлөм
  'wallet': Icons.account_balance_wallet_rounded,
  'coins': Icons.monetization_on_rounded,
  'banknote': Icons.payments_rounded,
  'credit-card': Icons.credit_card_rounded,
  'hand-coins': Icons.volunteer_activism_rounded,
  'percent': Icons.percent_rounded,

  // Банк, үнөмдөө, инвестиция
  'piggy-bank': Icons.savings_rounded,
  'landmark': Icons.account_balance_rounded,
  'trending-up': Icons.trending_up_rounded,
  'chart-pie': Icons.pie_chart_rounded,
  'chart-bar': Icons.bar_chart_rounded,
  'sprout': Icons.eco_rounded,

  // Пландоо жана эсеп
  'calculator': Icons.calculate_rounded,
  'receipt': Icons.receipt_long_rounded,
  'calendar': Icons.calendar_month_rounded,
  'clock': Icons.schedule_rounded,
  'target': Icons.track_changes_rounded,
  'scale': Icons.balance_rounded,

  // Коопсуздук
  'shield-check': Icons.verified_user_rounded,
  'key': Icons.key_rounded,
  'umbrella': Icons.umbrella_rounded,

  // Турмуш
  'home': Icons.home_rounded,
  'car': Icons.directions_car_rounded,
  'shopping-cart': Icons.shopping_cart_rounded,
  'gift': Icons.card_giftcard_rounded,
  'smartphone': Icons.smartphone_rounded,

  // Иш жана билим
  'briefcase': Icons.work_rounded,
  'building': Icons.apartment_rounded,
  'handshake': Icons.handshake_rounded,
  'graduation-cap': Icons.school_rounded,
  'book-open': Icons.menu_book_rounded,
  'lightbulb': Icons.lightbulb_rounded,

  // Мотивация
  'rocket': Icons.rocket_launch_rounded,
  'star': Icons.star_rounded,
  'flame': Icons.local_fire_department_rounded,
  'users': Icons.groups_rounded,
  'globe': Icons.public_rounded,
  'dumbbell': Icons.fitness_center_rounded,
};

/// The icon for [slug], or null when the slug is absent/unknown — the caller
/// then draws whatever it was going to draw anyway.
///
/// Kept for the places that genuinely need an [IconData] (a fallback, an
/// [IconButton]); anything that just draws the glyph should use
/// [lessonGlyph], which prefers the lucide artwork.
IconData? lessonIconFor(String? slug) =>
    slug == null ? null : _lessonIcons[slug];

/// The glyph for [slug] as a widget — lucide where we have it, the Material
/// equivalent where we do not, and [fallback] when the slug means nothing.
///
/// Every screen that paints a lesson, module, league or achievement icon goes
/// through here, so the two clients can never drift into drawing different
/// pictures for the same slug again.
Widget? lessonGlyph(
  String? slug, {
  double size = 24,
  Color? color,
  IconData? fallback,
}) {
  if (LucideIcon.has(slug)) return LucideIcon(slug!, size: size, color: color);
  final data = lessonIconFor(slug) ?? fallback;
  return data == null ? null : Icon(data, size: size, color: color);
}

File: shared/lessonIcons.js
// shared/lessonIcons.js — the lesson-node icon set.
//
// One list, four consumers: the admin panel renders it as a picker, the
// backend validates against it (contentStore.js#sanitizeLessonIcon), the web
// app maps each slug to a lucide component (shared/lessonIconComponents.jsx),
// and the Flutter app keeps a parallel map to Material icons
// (mobile/lib/src/core/lesson_icons.dart).
//
// Deliberately data-only — no React import — so admin-api can load it in
// Node without dragging a UI library into the server process.
//
// Adding an icon: append here, add the lucide component next to it in
// lessonIconComponents.jsx, and add the Material equivalent in
// lesson_icons.dart. A slug present here but missing from a client falls
// back to that client's default glyph rather than rendering nothing — but
// leaving one unmapped is a bug, not a feature.
//
// Slugs are permanent: a lesson stores the slug, so renaming one silently
// resets every lesson that used it back to the default icon.

export const LESSON_ICONS = [
  // Акча жана төлөм
  { slug: 'wallet',         label: 'Капчык' },
  { slug: 'coins',          label: 'Монеталар' },
  { slug: 'banknote',       label: 'Банкнот' },
  { slug: 'credit-card',    label: 'Карта' },
  { slug: 'hand-coins',     label: 'Акча берүү' },
  { slug: 'percent',        label: 'Пайыз' },

  // Банк, үнөмдөө, инвестиция
  { slug: 'piggy-bank',     label: 'Копилка' },
  { slug: 'landmark',       label: 'Банк' },
  { slug: 'trending-up',    label: 'Өсүш' },
  { slug: 'chart-pie',      label: 'Диаграмма' },
  { slug: 'chart-bar',      label: 'Графика' },
  { slug: 'sprout',         label: 'Өнүгүү' },

  // Пландоо жана эсеп
  { slug: 'calculator',     label: 'Калькулятор' },
  { slug: 'receipt',        label: 'Чек' },
  { slug: 'calendar',       label: 'Календарь' },
  { slug: 'clock',          label: 'Убакыт' },
  { slug: 'target',         label: 'Максат' },
  { slug: 'scale',          label: 'Тең салмак' },

  // Коопсуздук
  { slug: 'shield-check',   label: 'Коргоо' },
  { slug: 'key',            label: 'Ачкыч' },
  { slug: 'umbrella',       label: 'Камсыздандыруу' },

  // Турмуш
  { slug: 'home',           label: 'Үй' },
  { slug: 'car',            label: 'Унаа' },
  { slug: 'shopping-cart',  label: 'Соода' },
  { slug: 'gift',           label: 'Белек' },
  { slug: 'smartphone',     label: 'Телефон' },

  // Иш жана билим
  { slug: 'briefcase',      label: 'Жумуш' },
  { slug: 'building',       label: 'Компания' },
  { slug: 'handshake',      label: 'Келишим' },
  { slug: 'graduation-cap', label: 'Билим' },
  { slug: 'book-open',      label: 'Китеп' },
  { slug: 'lightbulb',      label: 'Идея' },

  // Мотивация
  { slug: 'rocket',         label: 'Старт' },
  { slug: 'star',           label: 'Жылдыз' },
  { slug: 'flame',          label: 'Стрик' },
  { slug: 'users',          label: 'Команда' },
  { slug: 'globe',          label: 'Дүйнө' },
  { slug: 'dumbbell',       label: 'Машыгуу' },
];

export const LESSON_ICON_SLUGS = new Set(LESSON_ICONS.map(i => i.slug));
